contracts/StakeLocker.sol

```diff
@@ -178,14 +182,15 @@ contract StakeLocker is StakeLockerFDT, Pausable {
     */
     function _updateStakeDate(address who, uint256 amt) internal {
         uint256 prevDate = stakeDate[who];
-        uint256 newDate  = block.timestamp;
-        if (prevDate == uint256(0)) {
-            stakeDate[who] = newDate;
-        } else {
-            uint256 dTime  = block.timestamp.sub(prevDate);
-            newDate        = prevDate.add(dTime.mul(amt).div(balanceOf(who) + amt));  // stakeDate + (now - stakeDate) * (amt / (balance + amt))
-            stakeDate[who] = newDate;
-        }
+        uint256 balance = balanceOf(who);
+
+        // stakeDate + (now - stakeDate) * (amt / (balance + amt))
+        // NOTE: prevDate = 0 implies balance = 0, and equation reduces to now
+        uint256 newDate = balance + amt > 0
+            ? prevDate.add(block.timestamp.sub(prevDate).mul(amt).div(balance + amt))
+            : prevDate;
+
+        stakeDate[who] = newDate;
```

Comment:
Wasn't this change in the other PR? Guess it still needs to get merged into this.

Reply:
Yup. Once I rebase on the other, it will be gone. I am not a git pro, so I don't really know how to avoid this.